### DISCUSSION TITLE
Why is ReqRes code split between network/ and sync/ ?

### DISCUSSION THREAD

**Discussion by dapplion** [ChainSafe/lodestar#1883] [General]
network/ contains a switch statement and implementations to deal with requests while sync/ contains a switch statement and implementations to deal with responses. Is there a specific reason to spread the logic in both directories?

**Comment by mpetrunic** [comment]
Because of modules codependency. To send request you just need peer id and network. To respond on rrquest you need sync, chain, db and others

**Reply by wemeetagain** [reply]
:+1: 
I'd always thought about it like:
- network code: define request/response interfaces, handle encoding/decoding
- sync code: link the network to the chain, handle responses that require the chain state

@dapplion If you can figure out a way to make it simpler, by all means go for it
